lib/client/client.go

```diff
@@ -1117,3 +1260,24 @@ func (proxy *ProxyClient) currentCluster() (*services.Site, error) {
 	}
 	return nil, trace.NotFound("cluster %v not found", proxy.siteName)
 }
+
+func (proxy *ProxyClient) sessionSSHCertificate(ctx context.Context, node string) (ssh.AuthMethod, error) {
+	if _, err := proxy.teleportClient.localAgent.GetKey(); trace.IsNotFound(err) {
+		// Either running inside the web UI in a proxy or using an identity
+		// file. Fall back to whatever AuthMethod we currently have.
+		return proxy.authMethod, nil
+	}
+	cluster, err := proxy.currentCluster()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	key, err := proxy.IssueUserCertsWithMFA(ctx, ReissueParams{
+		NodeName:       node,
+		RouteToCluster: cluster.Name,
+	})
```

Comment:
In order for this to work with the access request system, the request IDs need to be propagated (`ReissueParams.AccessRequests` field).  IDs of the currently active requests are stored on the SSH certificate.  #5441 introduces a helper method for extracting them, but you can do it manually with something like this:
```golang
var activeRequests services.RequestIDs
rawRequests, ok := sshCert.Extensions[teleport.CertExtensionTeleportActiveRequests]
if ok {
	if err := activeRequests.Unmarshal([]byte(rawRequests)); err != nil {
		return activeRequests, trace.Wrap(err)
	}
}
key, err := proxy.IssueUserCertsWithMFA(ctx, ReissueParams{
	AccessRequests: activeRequests.AccessRequests,
	// ...
})
```

Reply:
Thanks for the catch, added inside `IssueUserCertsWithMFA` to cover more callers.